compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -1068,3 +1098,212 @@ pub(super) enum LValue {
     MemberAccess { old_object: Values, index: usize, object_lvalue: Box<LValue> },
     Dereference { reference: Values },
 }
+
+#[derive(Default, Serialize, Deserialize)]
+pub(crate) struct GlobalsContext {
+    pub(crate) dfg: DataFlowGraph,
+
+    #[serde(skip)]
+    definitions: HashMap<LocalId, Values>,
+}
+
+impl GlobalsContext {
+    fn codegen_expression(&mut self, expr: &ast::Expression) -> Result<Values, RuntimeError> {
+        match expr {
+            ast::Expression::Ident(ident) => Ok(self.codegen_ident(ident)),
+            ast::Expression::Literal(literal) => self.codegen_literal(literal),
+            ast::Expression::Block(block) => self.codegen_block(block),
+            ast::Expression::Let(let_expr) => self.codegen_let(let_expr),
+            ast::Expression::Tuple(tuple) => self.codegen_tuple(tuple),
+            _ => {
```

Comment:
Really not a fan of duplicating all these codegen functions. Could we create a FunctionBuilder in SharedContext instead and reuse the existing code-gen functions? Would definitely need some setup code since we'd need a full context in that case not just a function builder, and we'd need to not recursively keep filling globals when we're already doing so.

I just see this code getting forgotten about very easily when we're updating any of these codegen functions. Don't want to have to worry about updating both versions each time.

Reply:
> Really not a fan of duplicating all these codegen functions

Yeah same here. I just duplicated for simplicity sake at the moment. I wanted to just use a `FunctionBuilder` but as you mentioned we do need a full context. I was planning to do this cleanup in followups but I can do it in this PR.

Reply:
I have re-used the existing builder flow. The set-up did not end up being that bad actually. Let me know what you think.